YAXLib/Caching/MemberWrapperCache.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (C) Sina Iravanian, Julian Verdurmen, axuno gGmbH and other contributors.
+// Licensed under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using YAXLib.Pooling.SpecializedPools;
+
+namespace YAXLib.Caching;
+
+/// <summary>
+///     Implements a singleton cache for <see cref="MemberWrapper"/>s
+///     to prevent creation of <see cref="MemberWrapper"/>s for the same type repetitively.
+///     <para>
+///     <b>Note</b>: The cache contains <b>all members</b> of a type that can <b>generally</b> be serialized.
+///     Filtering of undesired members takes place later in the de/serialization process.
+///     </para>
+/// </summary>
+internal class MemberWrapperCache : TypeCacheAbstract<IList<MemberWrapper>>
+{
+    public const int DefaultCacheSize = 1000;
```

Comment:
Is MaxCacheSize = DefaultCacheSize 1000 enough for most common use cases?

Reply:
Goes together with number of types to cache. 500 types with 20 members make 10,000 to cache.